### PR TITLE
Protobuf Changes

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
Changes to protobuffers according to Notion notes.

Signed-off-by: Georgi Yazovaliyski <georgi.yazovaliiski@gmail.com>

### CODE REVIEW

File: hapi-proto/src/main/proto/ResponseCode.proto [hunk 1]
```diff
@@ -214,7 +214,6 @@ enum ResponseCodeEnum {
   EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS = 200; // TokenTransfersTransactionBody has a TokenTransferList with no AccountAmounts
 
   INVALID_SCHEDULE_ID = 201; // The scheduled transaction entity is invalid or does not exist
-  INVALID_SCHEDULE_THRESHOLD = 202; // The scheduled transaction entity's threshold for execution is bigger than provided signatures
-  EMPTY_SIGNERS_LIST = 203; // No signers have been passed with the scheduled transaction entity
-  EMPTY_SIGNATURES = 204; // No signers have been passed with the scheduled transaction entity
+  EMPTY_SIGNATURES = 202; // No signers have been passed with the scheduled transaction entity
```
Comment (Daniel-K-Ivanov) [inline hapi-proto/src/main/proto/ResponseCode.proto:217]: `signers` -> `signatures`

File: hapi-proto/src/main/proto/ScheduleGetInfo.proto [hunk 2]
```diff
@@ -37,14 +37,13 @@ message ScheduleGetInfoQuery {
 }
 
 message ScheduleInfo {
-  ScheduleID schedule = 1; // The ID of the Scheduled Entity
+  ScheduleID scheduleID = 1; // The ID of the Scheduled Entity
```
Comment (Daniel-K-Ivanov) [inline hapi-proto/src/main/proto/ScheduleGetInfo.proto:40]: should be `schedule`. My bad..

File: hapi-proto/src/main/proto/ScheduleGetInfo.proto [hunk 3]
```diff
@@ -37,14 +37,13 @@ message ScheduleGetInfoQuery {
...
   AccountID schedulingAccount = 2; // The Account ID which created the Scheduled TX
-  bytes transactionBody = 3; // The transaction serialized into bytes that must be signed
-  Key adminKey = 4; // (optional) The Key which is able to delete the Scheduled Transaction (if tx is not already executed)
-  ThresholdAccount signers = 5; // The key which is able to authorise and sign the scheduled transaction. This could be ThresholdKey or a simple ed25519 key. Serves as definition to who should sign this scheduled TX in order for it to be executed.
-  SignatureMap sigMap = 6; // (optional) The signature map containing all signatures gathered so far.
-  bool executeImmediately = 7; // Always set to true in MVP!!!
-  Timestamp executeAtTime = 8; // The time at which the transaction will be attempted to be executed. Currently not supported.
+  AccountID payer = 3; // The account which is going to pay for the execution of the Scheduled TX
+  bytes transactionBody = 4; // The transaction serialized into bytes that must be signed
+  bool executeImmediately = 5; // Always set to true in MVP!!!
```
Comment (Daniel-K-Ivanov) [inline hapi-proto/src/main/proto/ScheduleGetInfo.proto:44]:
```
// Whether to execute the transaction immediately when all required signatures are collected. Currently, only executeImmediately=true is supported and will be returned
```

File: hapi-proto/src/main/proto/TransactionRecord.proto [hunk 4]
```diff
@@ -45,5 +45,6 @@ message TransactionRecord {
     }
     TransferList transferList = 10; // All hbar transfers as a result of this transaction, such as fees, or transfers performed by the transaction, or by a smart contract it calls, or by the creation of threshold records that it triggers.
     repeated TokenTransferList tokenTransferLists = 11; // All Token transfers as a result of this transaction
-    ScheduleID scheduleRef = 12; // Reference to the Scheduled TX
+    uint32 tieOrder = 12; // Order of events with the same consensus timestamp
+    ScheduleID scheduleRef = 13; // Reference to the Scheduled TX
```
Comment (Daniel-K-Ivanov) [inline hapi-proto/src/main/proto/TransactionRecord.proto:49]: `// Reference to the scheduled transaction ID that this transaction record represent`

File: hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java [hunk 5]
```diff
@@ -86,13 +86,15 @@ public ResponseCodeEnum validate(TransactionBody txnBody) {
             return validity;
         }
 
-        var signers = op.getSigners().getAccountsList();
-        if (signers.isEmpty()) {
-            return EMPTY_SIGNERS_LIST;
+        if (!op.hasPayer()) {
+            return MISSING_SCHEDULE_PAYER;
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java:90]: Payer is optional. It is stated in the comment describing it. No need to check

File: hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java [hunk 6]
```diff
@@ -86,13 +86,15 @@ public ResponseCodeEnum validate(TransactionBody txnBody) {
...
         }
-        if (signers.size() < op.getSigners().getThreshold()) {
-            return INVALID_SCHEDULE_THRESHOLD;
+
+        var sigMap = op.getSigMap();
+        if (op.hasSigMap() || sigMap.getSigPairList().isEmpty()) {
+            return EMPTY_SIGNATURES;
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java:95]: sigMap is optional. It is stated in the comment describing it. No need to check it

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

File: hapi-proto/src/main/proto/ResponseCode.proto [hunk 7]
```diff
@@ -214,6 +214,6 @@ enum ResponseCodeEnum {
   EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS = 200; // TokenTransfersTransactionBody has a TokenTransferList with no AccountAmounts
 
   INVALID_SCHEDULE_ID = 201; // The scheduled transaction entity is invalid or does not exist
-  EMPTY_SIGNATURES = 202; // No signers have been passed with the scheduled transaction entity
+  EMPTY_SIGNATURES = 202; // No signatures have been passed with the scheduled transaction entity
   MISSING_SCHEDULE_PAYER = 203; // No payer for scheduled transaction entity
```
Comment (Daniel-K-Ivanov) [inline hapi-proto/src/main/proto/ResponseCode.proto]: I think that those 2 are not used anymore?

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Review by failfmi — APPROVED** [review]

**Pull request merged**